### PR TITLE
OSD-14687: Add Backplane CLI managed jobs add,get,delete,log commands

### PR DESCRIPTION
**Pull request by samanthajayasinghe**
### What type of PR is this?

feature

### What this PR does / Why we need it?
This PR includes managed jobs create, delete. list and getting logs commands

### Which Jira/Github issue(s) does this PR fix?

_Resolves #_ [OSD-14687](https://issues.redhat.com/browse/OSD-14687)

### Special notes for your reviewer
You have to log in to the cluster before executing managed jobs commands 

### Pre-checks (if applicable)

- [x] Ran unit tests locally
- [x] Validated the changes in a cluster
- [x] Included documentation changes with PR

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,3 +2,76 @@
...
+## How backplane-cli works
+
+### Cloud Console
+
+Backplane supports logging into the cluster's cloud provider console. This makes it possible to perform operations such as debugging an issue, troubleshooting a customer misconfiguration, or directly access the underlying cloud infrastructure.
+
+When run `ocm backplane cloud console <CLUSTERID|EXTERNAL_ID|CLUSTER_NAME|CLUSTER_NAME_SEARCH>`, it will:
+
+- If the provided argument is a cluster name or an external cluster ID, retrieve the cluster ID and cluster name of the target cluster via OCM.
+- Lookup the backplane url if provided as a flag. If not specified, it defaults to the corresponding Hive shard of the target cluster via OCM.
```
Comment (wanghaoran1988) [inline README.md:16]:
```suggestion

```

File: README.md [hunk 2]
```diff
@@ -2,3 +2,76 @@
...
+### Managed Job
+
+A managed job is the running instance of a managed script in an OSD/ROSA cluster. Not to be confused with k8s job resource.
+
+Managed jobs run as pods in the `openshift-backplane-managed-scripts` namespace.
+
+When run `ocm backplane managedjob ...`, it will:
+
+- Retrive the logged-in cluster ID.
+- Lookup the corresponding Hive shard via OCM.
```
Comment (wanghaoran1988) [inline README.md:59]:
```suggestion

```

File: README.md [hunk 3]
```diff
@@ -2,3 +2,76 @@
...
+- Check if the cluster is not hibernating.
+- Create a managed job, `ocm backplane managedjob create <script_canonical_name>`:
+  - Script canonical name is vital to retrieve the relevant script from the [managed-scripts](https://github.com/openshift/managed-scripts) repo.
+  - Parses the canonical name and parameters(if any) to be sent as JSON payload to the backplane service.
+  - Sends a POST request to the backplane service: `POST https://<hive-shard>/backplane/script/{clusterId}/job`
```
Comment (wanghaoran1988) [inline README.md:64]:
```suggestion
  - Sends a POST request to the backplane service
```

File: README.md [hunk 4]
```diff
@@ -2,3 +2,76 @@
...
+  - The backplane service will validate the clusterID and the JSON payload and trigger the creation of a job pod and return a response body comprising of a message along with the job ID of the newly created managed job.
+- Fetch managed jobs, `ocm backplane managedjob get`:
+  - Retrieves all the managed jobs running in the namespace the backplane user has access to.
+  - If the job ID is provided as an argument, sends a GET request to the backplane service: `GET https://<hive-shard>/backplane/script/{clusterId}/job/{jobId}`
```
Comment (wanghaoran1988) [inline README.md:68]:
```suggestion
  - If the job ID is provided as an argument, sends a GET request to the backplane service: `GET https://<backplane-url>/backplane/script/{clusterId}/job/{jobId}`
```

File: README.md [hunk 5]
```diff
@@ -2,3 +2,76 @@
...
+  - If not then fetch all jobs, sends a GET request to the backplane service: `GET https://<hive-shard>/backplane/script/{clusterId}/job`
```
Comment (wanghaoran1988) [inline README.md:69]:
```suggestion
  - If not then fetch all jobs, sends a GET request to the backplane service: `GET https://<backplane-url>/backplane/script/{clusterId}/job`
```

File: README.md [hunk 6]
```diff
@@ -2,3 +2,76 @@
...
+  - Parses the response and displays managed job(s) running in the cluster.
+- Fetch managed job logs, `ocm backplane managedjob logs <jobID>`:
+  - Sends a GET request to the backplane service: `GET https://<hive-shard>/backplane/{clusterId}/job/{jobId}/logs`
```
Comment (wanghaoran1988) [inline README.md:72]:
```suggestion
  - Sends a GET request to the backplane service: `GET https://<backplane-url>/backplane/{clusterId}/job/{jobId}/logs`
```

File: README.md [hunk 7]
```diff
@@ -2,3 +2,76 @@
...
+  - Parses the response and displays the managed job logs.
+- Delete a managed job, `ocm backplane delete <jobName>`:
+  - Stops a running job and deletes all of its resources (including logs).
+  - Sends a DELETE request to thr backplane service: `DELETE https://<hive-shard>/backplane/{clusterId}/job/{jobId}`
```
Comment (wanghaoran1988) [inline README.md:76]:
```suggestion
  - Sends a DELETE request to thr backplane service: `DELETE https://<backplane-url>/backplane/{clusterId}/job/{jobId}`
```

File: cmd/ocm-backplane/managedJob/createManagedJob.go [hunk 8]
```diff
@@ -0,0 +1,148 @@
+/*
+Copyright © 2021 Red Hat, Inc
```
Comment (wanghaoran1988) [inline cmd/ocm-backplane/managedJob/createManagedJob.go:2]:
```suggestion
Copyright © 2023 Red Hat, Inc
```

File: cmd/ocm-backplane/managedJob/managedJob.go [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+	cmd := &cobra.Command{
+		Use:          "managedjob",
+		Aliases:      []string{"managedJob", "managedjob", "managedjobs"},
+		Short:        "Represents a backplane managedjob.",
+		SilenceUsage: true,
+	}
+
+	// url flag
+	// Denotes backplane url
+	// If this flag is empty, backplane-url will be populated by --cluster-id flag supplied by user. cluster-id flag will be used to find corresponding hive-shard and composing backplane url.
```
Comment (wanghaoran1988) [inline cmd/ocm-backplane/managedJob/managedJob.go:17]: This is not correct now

File: cmd/ocm-backplane/managedJob/managedJob.go [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+	cmd.PersistentFlags().String(
+		"url",
+		"",
+		"Specify backplane url. Default: The corresponding hive shard of the target cluster.",
```
Comment (wanghaoran1988) [inline cmd/ocm-backplane/managedJob/managedJob.go:21]:
```suggestion
		"Specify backplane url. ",
```

Comment (samanthajayasinghe) [reply cmd/ocm-backplane/managedJob/managedJob.go]: done

File: cmd/ocm-backplane/managedJob/managedJob.go (hunk 9 above)
Comment (samanthajayasinghe) [reply cmd/ocm-backplane/managedJob/managedJob.go]: done

File: cmd/ocm-backplane/managedJob/createManagedJob.go (hunk 8 above)
Comment (samanthajayasinghe) [reply cmd/ocm-backplane/managedJob/createManagedJob.go]: removed the code header since it's not consistence with other files

**Pull request merged**